esphome/components/online_image/bmp_image.cpp

```diff
@@ -0,0 +1,72 @@
+#include "bmp_image.h"
+
+#ifdef USE_ONLINE_IMAGE_BMP_SUPPORT
+
+#include "esphome/components/display/display.h"
+#include "esphome/core/helpers.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace online_image {
+
+static const char *const TAG = "online_image.bmp";
+
+int HOT BmpDecoder::decode(uint8_t *buffer, size_t size) {
+  size_t index = 0;
+  this->decoded_bytes_ += size;
+  if (this->current_index_ == 0 && this->decoded_bytes_ > 14) {
+    // Check if the file is a BMP image
+    if (buffer[0] != 'B' || buffer[1] != 'M') {
+      ESP_LOGE(TAG, "Not a BMP file");
+      return 0;
```

Comment:
Errors in online_image decoders are normally signaled with negative values
```suggestion
      return -1;
```